### ISSUE TITLE
Improve Vmc driver

### ISSUE THREAD

**Issue by gcarleo** [netket/netket#345]
We need a couple of improvements on the Vmc driver ```netket.Vmc``` : 

- add a ```run``` method with a similar interface to the one we previously had in the legacy ```netket.variational.Vmc``` 

- C-esque methods such as ```add_observable``` should be replaced with setters and getters for properties, such as 
```python
vmc = netket.Vmc(...)
vmc.observables={'sx':sx,...}
vmc.run(...) 
```

- we should think of allowing the i/o of run to be diverted to tensorboard or similar

**Comment by gcarleo** [comment]
Discussion about this started in #344

**Comment by PhilipVinc** [comment]
Closed by #353
